### ISSUE TITLE
Windows Terminal crashes when a profile's icon URL is unparseable/invalid

### ISSUE THREAD

**Issue by AdamDanischewski** [microsoft/terminal#2329]
I got the latest from choco and set up an Ubuntu terminal entry. I found that if I put a garbage URL in the profiles.json icon setting entry it crashes Windows Terminal and WT won't start thereafter. 

That probably shouldn't happen, though technically low priority its probably making WT look fragile. 

To repeat this behavior simply remove "ms-appx:" from any icon path.

**Comment by DHowett-MSFT** [comment]
Yes, this is a good followup to #1468

**Comment by AdamDanischewski** [comment]
#622 (**_There should be an error dialog for incorrect settings instead of just crashing_**) references #886 (**_Cannot run deployed application (winrt::hresult_error when debugging)_**), both are closed yet both underlying issues still linger. 

Sidenote: Tell MS to make Visual Studio free.

**Comment by DHowett-MSFT** [comment]
@AdamDanischewski 

>make Visual Studio free

Like [Community edition](https://visualstudio.microsoft.com/vs/community/)?

**Comment by AdamDanischewski** [comment]
Lol, yea, I meant Professional - I haven't used VS before but I see Community VS is full featured. I'm surprised these critical bugs aren't getting fixed quicker. 

I've been out of the loop for a few years touristing, interesting to see how far MS has embraced Open Source. Once I get a dev env set up I'll probably contribute.

**Comment by AdamDanischewski** [comment]
This issue is still not fixed as of v0.4.2382.0 - please re-open and de-merge this issue from #2376 since #2376 is fixed and this issue is not.

**Comment by DHowett-MSFT** [comment]
Interesting. I had to try about six different variants before I was able to find one that would crash.

**Comment by mdtauk** [comment]
There should be a default icon that a profile will revert to, when the specified icon is not found

**Comment by zadjii-msft** [comment]
> 
> 
> Interesting. I had to try about six different variants before I was able to find one that would crash.

@DHowett-MSFT _Mind sharing what URL variant caused the crash?_ Or filing a new issue with that variant?

**Comment by DHowett-MSFT** [comment]
```json
"icon": "/beeftown.png",
```

It can't construct a `Uri`.

```
Exception thrown at 0x00007FFCF5715369 (KernelBase.dll) in WindowsTerminal.exe:
WinRT originate error - 0x80070057 : '/beeftown.png is not a valid absolute URI.'.
```

**Comment by thomthom** [comment]
I just ran into this when I tried to use environmental variables in the icon path.

This will crash in 0.5.2681.0
```
"icon": "%HOMEPATH%\\Dropbox\\terminal\\VSWinIcon_100x.png",
```

This will not.
```
"icon": "C:\\Users\\Thomas\\Dropbox\\terminal\\VSWinIcon_100x.png",
```

I realise that env variables might not be supported for icons. But just wanted to throw in my crash scenario.

**Comment by mkitzan** [comment]
Does this issue still repro? I can't get a crash with any of the icon paths. Let me know if you've got an icon path which can consistently crash WT, and any specific repro steps required.

**Comment by Reelix** [comment]
I posted in #2376 which got merged into this issue. Using my original steps, I am unable to reproduce this issue in 0.7.3451.0 so it seems to have been fixed.

**Comment by mkitzan** [comment]
Phantom fixes, nice. Would be good if others in the thread who had repros after #2376 was merged could confirm too.

**Comment by thomthom** [comment]
I'm not able to reproduce this either now.

**Comment by DHowett-MSFT** [comment]
Fascinating. I'm going to close this for now, but we'll keep our eyes peeled. Thanks for the confirmations!

**Comment by zadjii-msft** [comment]
We're reactivating this from discussion in #4002:


> 
> ```
> Windows build number: Microsoft Windows NT 10.0.19041.0
> Windows Terminal version (if applicable): 0.7.3451.0
> ```
> 
> # Steps to reproduce
> 
> In `profiles.json` under a profile, add a background image path that references the home directory `~`.
> 
> ```json
> {
>     ...
>     "profiles": [
>         {
>             "name": "Windows PowerShell",
>             "backgroundImage": "~/Pictures/Console/PowerShell.png",
>         }
>     ...
> }
> ```
> 
> # Expected behavior
> 
> Should not crash. Should show the expected background image.
> # Actual behavior
> 
> Windows Terminal crash at startup.

**Comment by mkitzan** [comment]
The particular instance of this crash from #4002 is due to [`SelectionBackground()`](https://github.com/microsoft/terminal/blob/master/src/cascadia/TerminalControl/TermControl.cpp#L272) in the `UpdateSettings` event throwing due to the file path.

Rather than defensively catching an exception in the async handler, I wonder if it would make sense to validate the media associated with a profile somewhere during the profile creation process. Then we could intercept the issue (error page, inserting some default media...) before the `UpdateSettings` event is even launched.
